lua/lazyvim/plugins/editor.lua

```diff
@@ -161,48 +161,47 @@ return {
         desc = "Goto Symbol (Workspace)",
       },
     },
-    opts = {
-      defaults = {
-        prompt_prefix = " ",
-        selection_caret = " ",
-        mappings = {
-          i = {
-            ["<c-t>"] = function(...)
-              return require("trouble.providers.telescope").open_with_trouble(...)
-            end,
-            ["<a-t>"] = function(...)
-              return require("trouble.providers.telescope").open_selected_with_trouble(...)
-            end,
-            ["<a-i>"] = function()
-              local action_state = require("telescope.actions.state")
-              local line = action_state.get_current_line()
-              Util.telescope("find_files", { no_ignore = true, default_text = line })()
-            end,
-            ["<a-h>"] = function()
-              local action_state = require("telescope.actions.state")
-              local line = action_state.get_current_line()
-              Util.telescope("find_files", { hidden = true, default_text = line })()
-            end,
-            ["<C-Down>"] = function(...)
-              return require("telescope.actions").cycle_history_next(...)
-            end,
-            ["<C-Up>"] = function(...)
-              return require("telescope.actions").cycle_history_prev(...)
-            end,
-            ["<C-f>"] = function(...)
-              return require("telescope.actions").preview_scrolling_down(...)
-            end,
-            ["<C-b>"] = function(...)
-              return require("telescope.actions").preview_scrolling_up(...)
-            end,
-          },
-          n = {
-            ["q"] = function(...)
-              return require("telescope.actions").close(...)
-            end,
+    opts = function()
+      local actions = require("telescope.actions")
+
+      local open_with_trouble = function(...)
+        return require("trouble.providers.telescope").open_with_trouble(...)
+      end
+      local open_selected_with_trouble = function(...)
+        return require("trouble.providers.telescope").open_selected_with_trouble(...)
+      end
```

Comment:
You may want to leave these and the insert mode trouble.nvim maps as anonymous functions to keep the lazy-loading.

Reply:
Lazy loading shouldn't be affected though, since I'm requiring the modules inside the function bodies.

Reply:
I've also changed `opts` into its function form, which `lazy` will invoke until loading Telescope, so this shouldn't do any premature loading.

Reply:
> Lazy loading shouldn't be affected though, since I'm requiring the modules inside the function bodies.

I guess I'm mistaken, when I made the same changes locally, it seemed like Trouble with being loaded by Telescope.

Reply:
I gosh, yeah I'm sorry, I misread the code.

Reply:
@jyuan0 oh my don't sweat it, it happens to me all the time <3
Thank you for the review!

Reply:
Yes, this is the same code essentially, expect that it assigns a function name. Should have no effect on lazy loading. Looking forward to this PR. I always looked at those anonymous functions, but got used to them (by reading about the keymaps from the source code) and never bothered with them. Thank you for this.